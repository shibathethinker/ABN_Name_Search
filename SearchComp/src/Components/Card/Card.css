.cardBox{
    border-color: #ecf0f1 !important;
    border-width: 1px;
    border-style: solid;
    display: flex;
    width:90%;
    border-radius: 6px;
    box-shadow: 1px 2px #91bfc3;
    margin: auto;
    margin-top: 10px;
    flex-direction: column;
    font-size:14px;
    align-items: center;
    color: #2C3E50;
    font-family: 'Candara';
}

.cardBox:hover{
    transform: scale(1.1);
    background-color: #ecf0f1;
}

.cardBox label{

    color:#2C3E50;
    font-weight: bold;
    border-radius: 2px;

}
.cardBox div{
    margin:2px;
}

.cardBox h2{
    width:100%;
     cursor: pointer;

}

.cardBox:hover h2{
    text-decoration: underline;
}

.cardBoxSectionInvisible{
    display: none;
}

.cardBoxSectionVisible{
    
}

.cardBox .cardBoxSection{
    display: flex;
    margin: auto;
    display: -webkit-flex;
    display: flex;
    -webkit-flex-direction: row;
    flex-direction: row;
    -webkit-align-content: center;
    align-content: center;
    justify-content: space-between;
}


@media (min-width:640px){
.cardBox{
    width:50%
}

}

