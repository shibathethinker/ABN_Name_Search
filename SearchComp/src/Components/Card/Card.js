import React from 'react';
import './Card.css';

export default class Card extends React.Component{

    constructor(props)
    {
        super(props);
        this.toggleView=this.toggleView.bind(this);
        this.state={visible:false}
        this.allProps={...this.props.allProps}
        this.displayContent='';
    }

    toggleView(e)
    {
      this.setState({visible:!this.state.visible})
    }

    render()
    {
      let name=this.allProps.Name===undefined?this.allProps.EntityName:this.allProps.Name;

      this.displayContent=[];
  for(let prop in this.allProps)
    {
      let val=this.allProps[prop];
    this.displayContent.push(<div><label>{prop} : </label>{val}</div>)
    }

            return (
             <div className="cardBox">
<h2 onClick={this.toggleView}>{name}</h2>

<div  className={!this.state.visible?"cardBoxSectionInvisible":""}>
{this.displayContent}
</div>
                </div>
                )

    }


}