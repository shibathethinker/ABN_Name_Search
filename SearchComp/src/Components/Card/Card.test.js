import React from 'react';
import Enzyme,{shallow,mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Card from './Card'
import ShallowRenderer from 'react-test-renderer/shallow'

Enzyme.configure({adapter:new Adapter()});

test('Card renders properly',()=>{
    const wrapper=shallow(<Card></Card>);
    expect(wrapper.length).toBe(1);
})

test('Checking Card Snapshot',()=>{
let renderer=new ShallowRenderer();
let instance=renderer.render(<Card/>)
expect(instance).toMatchSnapshot();
})