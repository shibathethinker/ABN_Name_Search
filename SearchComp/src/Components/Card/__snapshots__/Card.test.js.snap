// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking Card Snapshot 1`] = `
<div
  className="cardBox"
>
  <h2
    onClick={[Function]}
  />
  <div
    className="cardBoxSectionInvisible"
  />
</div>
`;
